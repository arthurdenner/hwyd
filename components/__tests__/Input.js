import React from 'react';
import { Text } from 'react-native';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Input } from '../';
import theme from '../../utils/theme';

describe('<Input />', () => {
  it('renders an email input', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Input
          label='Email Address'
          type="email"
          placeholder='you@domain.com'
        />
      </ThemeProvider>,
    );
    const root = tree.find(Input).shallow();
    expect(root).toMatchSnapshot();
  });

  it('renders a password input', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Input
          label='Password'
          type="password"
          autoCorrect={false}
          placeholder='*******'
          secureTextEntry
        />
      </ThemeProvider>,
    );
    const root = tree.find(Input).shallow();
    expect(root).toMatchSnapshot();
  });
});
