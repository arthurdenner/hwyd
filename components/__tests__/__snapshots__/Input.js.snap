// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders a password input 1`] = `
<Styled(View)>
  <Styled(Text)>
    PASSWORD
  </Styled(Text)>
  <Styled(TextInput)
    autoCorrect={false}
    placeholder="*******"
    secureTextEntry={true}
    type="password"
  />
</Styled(View)>
`;

exports[`<Input /> renders an email input 1`] = `
<Styled(View)>
  <Styled(Text)>
    EMAIL ADDRESS
  </Styled(Text)>
  <Styled(TextInput)
    autoCorrect={false}
    placeholder="you@domain.com"
    type="email"
  />
</Styled(View)>
`;
